File: solutions/lua/square-root/1/square-root.lua
local SquareRoot = {}

function SquareRoot.square_root(radicand)
	local r = radicand;
	local err = 0E-10
	while (math.abs(radicand - r*r) > err) do
		r = (r + (radicand / r)) / 2
	end
	return r
end

return SquareRoot

File: solutions/ocaml/square-root/1/square_root.ml
open Base

let square_root (radicand : int) : int =
  let radicand = Float.of_int radicand in
  let err = 0E-10 in
  let rec aux (r : float) : float =
    if Float.((abs (radicand - r*r)) <= err) then
      r
    else
      aux Float.((r + (radicand / r)) / 2.0)
  in Float.to_int (aux (radicand))
